Marlin/src/module/stepper.cpp

```diff
@@ -2862,18 +2892,185 @@ hal_timer_t Stepper::block_phase_isr() {
 }
 
 #if ENABLED(LIN_ADVANCE)
+  #if ENABLED(SMOOTH_LIN_ADV)
+    void Stepper::set_la_interval(int32_t rate) {
+      if (rate == 0) {
+        la_interval = LA_ADV_NEVER;
+      }
+      else {
+        const bool forward_e = rate > 0;
+        la_interval = calc_timer_interval(uint32_t(ABS(rate)));
+        if (forward_e != motor_direction(E_AXIS)) {
+          last_direction_bits.toggle(E_AXIS);
+          count_direction.e = -count_direction.e;
+          DIR_WAIT_BEFORE();
+          E_APPLY_DIR(forward_e, false);
+          TERN_(FTM_OPTIMIZE_DIR_STATES, last_set_direction = last_direction_bits);
+          DIR_WAIT_AFTER();
+        }
+      }
+    }
+    #if ENABLED(INPUT_SHAPING_E_SYNCH)
+      constexpr uint16_t IS_COMPENSATION_BUFFER_SIZE = 
+        (SMOOTH_LIN_ADV_HZ / (20 /* min input shaping frequency*/) / 2.0f + 0.5f);
+      typedef struct {
+          float x;
+          float y;
+      } XYPair;
+
+      typedef struct {
+          XYPair buffer[IS_COMPENSATION_BUFFER_SIZE];
+          uint16_t index;
+      } DelayBuffer;
+
+      DelayBuffer delayBuffer;
+
+      void add_to_buffer(XYPair input) {
+        delayBuffer.buffer[delayBuffer.index] = input;
+        delayBuffer.index++;
+        if (delayBuffer.index == IS_COMPENSATION_BUFFER_SIZE) {
+          delayBuffer.index = 0;
+        }
+      }
+
+      XYPair lookback(shaping_time_t t /* in stepper timer ticks */) {
+        constexpr float STEPPER_TICKS_PER_ADV_TICS = (float) SMOOTH_LIN_ADV_HZ / STEPPER_TIMER_RATE;
```

Comment:
I think the variable name `STEPPER_TICKS_PER_ADV_TICS` is the wrong way round. Shouldn't it be `ADV_TICS_PER_STEPPER_TICKS`?

Reply:
good catch